code/modules/mob/living/carbon/slime/powers.dm

```diff
@@ -226,4 +226,16 @@
 		else
 			src << "<i>I am not ready to reproduce yet...</i>"
 	else
-		src << "<i>I am not old enough to reproduce yet...</i>"
\ No newline at end of file
+		src << "<i>I am not old enough to reproduce yet...</i>"
+
+/mob/living/carbon/slime/verb/SetSlimeName()
+	set name = "Set Name"
+	set category = "Slime"
+	set desc = "This will change your name."
+
+	var/OldSlimeName = name
+	var/NewSlimeName = copytext(sanitize(input(usr, "Put your gooey title here.", "Pick a name!", OldSlimeName)),1,MAX_NAME_LEN)
```

Comment:
That is what we have game admins for Imgoofball. Silmes, golems and others can also talk through a turned on intercom/radio.

The code here is a little messy though.